src/transformers/trainer.py

```diff
@@ -1846,13 +1852,15 @@ def training_step(self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor,
         inputs = self._prepare_inputs(inputs)
 
         if is_sagemaker_mp_enabled():
-            scaler = self.scaler if self.use_amp else None
+            scaler = self.scaler if self.do_grad_scaling else None
             loss_mb = smp_forward_backward(model, inputs, self.args.gradient_accumulation_steps, scaler=scaler)
             return loss_mb.reduce_mean().detach().to(self.args.device)
 
         if self.use_amp:
-            with autocast():
+            with autocast(dtype=self.amp_dtype):
                 loss = self.compute_loss(model, inputs)
+                print(f"{self.amp_dtype}")
```

Comment:
I think these two should be calls to call `logger.info` and should include more than just the dtype or loss, otherwise, users are forced to read the source code to figure out what these messages mean.

Reply:
that was just a debug left-over - thank you for catching that, @manuelciosici! will remove